geoengine/auth.py

```diff
@@ -55,6 +56,11 @@ def __init__(self, server_url: str, credentials: Tuple[str, str] = None) -> None
             session = req.post(f'{server_url}/login',
                                json={"email": os.environ.get("GEOENGINE_EMAIL"),
                                      "password": os.environ.get("GEOENGINE_PASSWORD")}).json()
+        elif token is not None:
+            session = req.get(f'{server_url}/session', headers={'Authorization': f'Bearer {token}'}).json()
+        elif "GEOENGINE_TOKEN" in os.environ:
+            session = req.get(f'{server_url}/session',
+                              headers={'Authorization': f'Bearer {os.environ.get("GEOENGINE_TOKEN")}'}).json()
```

Comment:
Shouldn't it be the other way around? If there is a token, use it. If not and you have credentials, use them.

Otherwise, we should raise an error if there are both set.

Reply:
b03e5db6ca83c5c6f3edaad7797b3ac98087f558